### ISSUE TITLE
Using --check in `sqlfluff fix` throws an OSError

### ISSUE THREAD

**Issue by tiffanygwilson** [sqlfluff/sqlfluff#6969]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I was having an issue with fixes not being applied, so I wanted to try using `--check` with `sqlfluff fix`:

```
sqlfluff fix --check models/sqlfluff_playground.sql
```

stdout displayed the "are you sure" prompt but errored out immediately thereafter.

### Expected Behaviour

I expected to be able to enter y/n after the confirmation prompt

### Observed Behaviour

```shell
==== finding fixable violations ====
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models/reproduce_error.sql] FAIL
L:   1 | P:   1 | LT09 | Select targets should be on a new line unless there is
                       | only one select target. [layout.select_targets]
L:   3 | P:   1 | LT09 | Select targets should be on a new line unless there is
                       | only one select target. [layout.select_targets]
L:   3 | P:  36 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
==== fixing violations ====
3 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] Traceback (most recent call last):
  File "/usr/src/app/.venv/bin/sqlfluff", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/sqlfluff/cli/commands.py", line 1134, in fix
    _paths_fix(
  File "/usr/src/app/.venv/lib/python3.11/site-packages/sqlfluff/cli/commands.py", line 940, in _paths_fix
    c = click.getchar().lower()
        ^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/termui.py", line 744, in getchar
    return _getchar(echo)
           ^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/_termui_impl.py", line 781, in getchar
    with raw_terminal() as fd:
  File "/usr/local/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/click/_termui_impl.py", line 759, in raw_terminal
    f = open("/dev/tty")
        ^^^^^^^^^^^^^^^^
OSError: [Errno 6] No such device or address: '/dev/tty'
```

### How to reproduce

sql:
```sql
SELECT "a" AS a, "b" AS b, "c" AS c
UNION ALL
SELECT "d" AS a, "e" AS b, "f" AS c
```

command:
```shell
sqlfluff fix --check models/sqlfluff_playground.sql
```

### Dialect

bigquery

### Version

From verbose linting output:
```
==== sqlfluff ====
sqlfluff:                3.3.1 python:                3.11.11
implementation:        cpython verbosity:                   4
dialect:              bigquery templater:                 dbt
dbt:                    =1.9.2
rules:                                all
```

### Configuration

.sqlfluff (not in great shape because of troubleshooting attempts)
```ini
[sqlfluff]
dialect = bigquery
# See https://docs.sqlfluff.com/en/stable/configuration/templating/dbt.html
templater = dbt
ignore_templated_areas = False

# Comma separated list of rules to exclude, or None
# See https://docs.sqlfluff.com/en/stable/configuration.html#enabling-and-disabling-rules
# AM04 (ambiguous.column_count) and ST06 (structure.column_order) are
# two of the more controversial rules included to illustrate usage.
exclude_rules = structure.using, structure.column_order, references.from, convention.terminator, structure.constant_expression, AL06, AM01, ST05, CV12, RF02, LT01

# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
# Set to zero or negative to disable checks.
max_line_length = 175

# CPU processes to use while linting.
# The default is "single threaded" to allow easy debugging, but this
# is often undesirable at scale.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specified_number.
# e.g. -1 means use all processors but one. 0 means all cpus.
processes = -1

# Warn instead of fail on errors, mainly to use during transition.
warnings = LT02, CP01, CV01, CV04

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
load_macros_from_path = macros
library_path = dbt_packages

[sqlfluff:templater:dbt:context]
partition_run_id = 20250601

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-indent-locations
indent_unit = space
tab_space_size = 2
indented_ctes = True
indented_joins = False
indented_using_on = True
indented_on_contents = False
indented_then = False
indented_then_contents = True
allow_implicit_indents = False
template_blocks_indent = True
# This is a comma seperated list of elements to skip
# indentation edits to.
skip_indentation_in = script_content
# If comments are found at the end of long lines, we default to moving
# them to the line _before_ their current location as the convention is
# that a comment precedes the line it describes. However if you prefer
# comments moved _after_, this configuration setting can be set to "after".
trailing_comments = before


# Layout
[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = none

# Will be available when dbt base image updates to sqlfluff >= 3.4.0.
# Use ./local_run_2 sqlfluff --version to check the version.
# [sqlfluff:rules:layout.newlines]
# maximum_empty_lines_inside_statements = 2

[sqlfluff:layout:type:common_table_expression]
keyword_line_position = leading

# Aliasing
[sqlfluff:rules:aliasing.table]
aliasing = explicit

[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:aliasing.length]
min_alias_length = 3


# Capitalisation
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower
unquoted_identifiers_policy = aliases
# extended_capitalisation_policy = snake

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

# Other conventions
[sqlfluff:rules:convention.select_trailing_comma]
select_clause_trailing_comma = none

[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = ansi

[sqlfluff:rules:convention.terminator]
multiline_newline = True
require_final_semicolon = False

[sqlfluff:rules:ambiguous.join]
fully_qualify_joint_types = both

```

.sqlfluffignore
```
# folders to ignore (we only want to check models/ and tests/)
target/
dbt_packages/
macros/
dbt-packages-local-artifactory/
.vscode-server/
dbt_docs/
dbt_packages/
docs/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Given the path in the traceback, I'm assuming this is in a docker container. Do you have the exact command this is running with?

On the docker page I know that this is recommended to be setup like so:

```sh
alias sqlfluff='docker run -it --rm -v $PWD:/sql sqlfluff/sqlfluff:<version_tag>'
```

**Comment by tiffanygwilson** [comment]
Ooh good point. Unfortunately we're executing a plain `sqlfluff fix` after copying the `.sqlconfig` to the container. Thanks for the reference to the [Docker page](https://hub.docker.com/r/sqlfluff/sqlfluff); I will follow up with the team that owns how everything is executed in the container.

In the meantime, can you please explain why or how this setup may fix the error I encountered? I know the team that owns this will ask. :)

**Comment by keraion** [comment]
The lack of a tty would have me believe that `docker run` is lacking the [`-t` option](https://docs.docker.com/reference/cli/docker/container/run/#tty)

**Comment by tiffanygwilson** [comment]
And you would be correct. Thank you!

**Comment by tiffanygwilson** [comment]
@keraion thanks again, I confirmed that adding the `-it` flag to `docker run` fixed the problem.
